### PR TITLE
#164693398 Add the feature for users to be able to get all articles with pagination

### PR DESCRIPTION
**Pull request by codinger41**
#### What does this PR do?
Adds the feature for users to be able to search and filter articles.
#### Description of Task to be completed?
Users should be able to get articles with pagination.
Users should be able to get  5 highest rated articles.
Users should be able to get 5 highest commented articles.
This new endpoint takes these queries:
- `limit` which is the number of articles to return, the default is 10.
- `offset` which is the number of articles to skip. the default is 0.

#### How should this be manually tested?
- Make a GET request to `/api/v1/articles` and add query parameters `limit` and `offset` with numbers as required.

- Make a GET request to  `/api/v1/articles/order` with the queries `type` which is the type of order, it can be `comments`, `ratings` or `latest`. 

#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/164693398

#### Questions:
N/A

### CODE REVIEW

**Comment by Dubby20** [comment]
Format your PR message

File: controllers/articles.js [hunk 1]
```diff
@@ -267,4 +281,125 @@ export default class ArticleController {
...
+            .slice(0, amount);
+        } else {
+          articles = results
+            .sort((a, b) => b.Ratings.length - a.Ratings.length)
+            .slice(0, amount);
+        }
+      }
+      return res.status(200).json({
+        success: true,
+        articles
```
Comment (Dubby20) [inline controllers/articles.js]: No response message

Comment (codinger41) [reply controllers/articles.js]: There's no need for a message if the request was successful @Dubby20

**Review by kleva-j — APPROVED** [review]

File: controllers/articles.js (hunk 1 above)
Comment (kleva-j) [reply controllers/articles.js]: If a request is successful you have to send a message to let the user know the status of their request.

**Review by Dubby20 — APPROVED** [review]

**Review by olajide1234 — APPROVED** [review]

File: controllers/articles.js [hunk 2]
```diff
@@ -267,4 +281,137 @@ export default class ArticleController {
...
+            },
+            {
+              model: Ratings,
+            },
+          ],
+        });
+      } else {
+        articles = await Article.findAll({
+          where: {},
+          order: [[Comment, 'comment', 'ASC']],
```
Comment (9jaswag) [inline controllers/articles.js]: @leksyib have you tested this to ensure it actually returns the result in the right order?

Comment (codinger41) [reply controllers/articles.js]: Yes I have @9jaswag

Comment (9jaswag) [reply controllers/articles.js]: @leksyib  I'm surprised that `ASC` returns the right data. What does `DESC` return?

Comment (codinger41) [reply controllers/articles.js]: `DESC` returns the data with the lowest below the highest. So the ones with the highest ratings come first and the lowest will come after

File: controllers/articles.js [hunk 3]
```diff
@@ -267,4 +281,112 @@ export default class ArticleController {
...
+  }
+
+  /**
+  * @description - Get a specific number of articles with criteria
+  * @static
+  * @param {Object} type - type of order
+  * @memberof ArticleController
+  * @returns {Object} order for findAll
+  */
+  static getOrder(type) {
```
Comment (9jaswag) [inline controllers/articles.js]: this is slightly better. There are better alternatives but it sure beats the previous `if/else` logic.

File: controllers/articles.js [hunk 4]
```diff
@@ -273,4 +287,112 @@ export default class ArticleController {
...
+
+  /**
+  * @description - Get a specific number of articles with criteria
+  * @static
+  * @param {Object} req - the request object
+  * @param {Object} res - the response object
+  * @memberof ArticleController
+  * @returns {Object} class instance
+  */
+  static async getArticlesByHighestField(req, res) {
```
Comment (9jaswag) [inline controllers/articles.js:390]: @leksyib no change has been made here

**Review by 9jaswag — CHANGES_REQUESTED** [review]

File: controllers/articles.js [hunk 5]
```diff
@@ -273,4 +287,121 @@ export default class ArticleController {
...
+      latest:
+      `
+        SELECT *
+        FROM "Articles"
+        ORDER BY "Articles"."createdAt" DESC
+        LIMIT ${Number(amount) || 5}
+      `,
+      comments:
+      `
+        SELECT "Articles".*, COUNT("Comments".comment) AS avg_count
```
Comment (9jaswag) [inline controllers/articles.js]: @leksyib this should be `comment_count`. We're not running an average on the count.

**Review by 9jaswag — CHANGES_REQUESTED** [review]

File: controllers/articles.js (hunk 5 above)
Comment (codinger41) [reply controllers/articles.js]: Thanks I'll change that @9jaswag

Comment (9jaswag) [reply controllers/articles.js]: @leksyib this hasn't been changed yet

Comment (codinger41) [reply controllers/articles.js]: @9jaswag  changed now

**Review by 9jaswag — APPROVED** [review]

**Pull request merged**